A41-xds-rbac.md

```diff
@@ -0,0 +1,199 @@
+xDS-enabled gRPC Servers, and support its `RBACPerRoute` configuration override.
+This does _not_ include [RBAC network filter][] support nor running the filter
+on gRPC clients. xDS v2 support is not necessary.
+
+Supporting the RBAC HTTP filter on server-side leverages [A36: xDS-Enabled
+Servers][A36] and [A39: xDS HTTP Filter Support][A39]. Server-side HTTP filters
+are less common than client-side at this point in time, so this may be the first
+usage of `RouteConfiguration` on server-side and so likely involves adding more
+complete support for A39 by creating+executing filters on server-side and
+processing `VirtualHost`s and `RoutMatch`es to determine which configuration
+should be provided to each filter. A39 should be consulted for the expected
+behavior.
+
+The core policy matching logic should be split into an "RBAC engine" to allow
+reuse with non-xDS environments. However, there is no requirement that the RBAC
+engine have a specialized API; it could simply be an interceptor.
+
+The [RBAC policy][] has many fields. All current (Envoy-implemented) fields will
+be considered in gRPC. However, some fields may have pre-determined values or
+behavior. At this time, if the `RBAC.action` is `Action.LOG` then the policy
+will be completely ignored, as if RBAC was not configurated. CEL fields are not
+supported, so if `Policy.condition` or `Policy.checked_condition` is present the
+XdsClient must NACK the configuration.
+
+The following fields of `Permission` may not be obvious how they map to gRPC:
+
+| Permission Field | gRPC Equivalent |
+| ---------------- | --------------- |
+| header           | Metadata (with caveats) |
+| url_path         | Fully-qualified RPC method name with leading slash |
+| destination_ip   | Local address for this connection |
+| destination_port | Local port for this connection |
+| metadata         | Hard-coded as empty; never matches |
+| requested_server_name | Hard-coded as empty string |
+
+Be strongly aware that Envoy Metadata has no relation to gRPC's Metadata.
+Because matching supports NOT, the matcher must still be processed even if a
+rule contains references to things that don't generally match; it is not trivial
+to "optimize out" the never-matching rules.
+
+The `header` field is not entirely 1:1 with gRPC Metadata. To begin with, gRPC
+Metadata isn't 100% consistent cross-language in its handling of hop-to-hop
+headers (e.g., `TE`) and pseudo-headers. For this design, `headers` can include
+`:method`, `:scheme`, `:authority`, and`:path` matchers and they must match the
+values received on-the-wire independent of whether they are stored in Metadata
+or in separate APIs. It is unspecified whether hop-to-hop headers are matched.
+Multi-valued metadata is represented as the concatenation of the values along
+with a `,` (comma, no added spaces) separator, as permitted by HTTP and gRPC.
+Binary headers are represented in their base64-encoded form, although we rarely
```

Comment:
I'm wondering if we should just exclude headers ending in `-bin` here (i.e., pretend they don't exist).  That would be consistent with how we handle them in route matching.